packages/ui-components/src/components/MenuItem/MenuItem.component.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useContext, MouseEvent, FC } from "react"
+import { Menu as HLMenu } from "@headlessui/react"
+import { MenuContext } from "../Menu/Menu.component"
+import { Icon, KnownIconsEnum } from "../Icon/Icon.component"
+
+const itemStyles = `
+   jn-text-theme-default
+   jn-flex
+   jn-items-center
+   jn-w-full
+   cursor-pointer
+   bg-clip-padding
+   jn-truncate
+   jn-text-left
+   jn-bg-theme-background-lvl-1
+   disabled:jn-cursor-not-allowed
+   data-[headlessui-state="disabled"]:jn-cursor-not-allowed
+`
+
+const smallStyles = `
+   jn-text-sm
+   jn-p-2
+`
+
+const normalStyles = `
+   jn-text-base
+   jn-pt-[0.6875rem]
+   jn-pb-[0.5rem]
+   jn-px-[0.875rem]
+`
+
+const actionableItemStyles = `
+   hover:jn-bg-theme-background-lvl-3
+   data-[headlessui-state="disabled"]:jn-bg-theme-background-lvl-3
+`
+
+export interface MenuItemProps {
+  /** Children of the menu item */
+  children?: React.ReactNode
+  /** Pass a custom className to the menu item */
+  className?: string
+  /** Whether the menu item is disabled */
+  disabled?: boolean
+  /** Pass the name of an icon the button should show. Can be any icon included with Juno. */
+  icon?: keyof typeof KnownIconsEnum
+  /** The label of the menu item. Will take precedence over children passed to the component. */
+  label?: string
+  /** Pass an href to the menu item. Will result in the menu item being rendered as an `<a>`. */
+  href?: string
+  /** Pass an onClick handler to the menu item. Will result in the menu item being rendered as a `<button>`. */
+  onClick?: (_event: MouseEvent<HTMLButtonElement>) => void
+}
+
+interface MenuContextType {
+  variant: "small" | "normal"
+}
+
+/** 
+ A menu item to be used inside Menu.
+ Can render `<a>`, `<button>`, or `<div>` based on props.
+ */
+export const MenuItem: FC<MenuItemProps> = ({
+  children = null,
+  className = "",
+  disabled = false,
+  href = "",
+  icon = null,
+  label = "",
+  onClick = undefined,
+  ...props
+}) => {
+  const menuContext = useContext(MenuContext) as MenuContextType | null
+  const variant = menuContext?.variant ?? "normal"
+
+  // Determine which element to render:
+  const Element = href ? "a" : children ? "div" : onClick ? "button" : "div"
```

Comment:
Since `div` is the default if neither `href` nor `onClick` applies,  couldn't this be simplified to this?
```suggestion
  const Element = href ? "a" : onClick ? "button" : "div"
```

Reply:
That's what @franzheidl tried, but it was causing a bug so I fixed it by re-introducing https://github.com/cloudoperators/juno/pull/575/commits/48c526fbb65d2cd33e6bab0c3ca4a27b9cf1a6d8

Reply:
This is the solution @guoda-puidokaite found to prevent storybook from wrapping their doing own onClick handler magic, resulting in the componentn always rendering a button. 

Since this is only an issue in the standalone `MenuItem` story (which will never occur IRL)  and we may want 
to get rid of these anyway, I'd suggest going with this solution for now.

Reply:
Regarding [this](https://github.com/cloudoperators/juno/pull/575#issuecomment-2457559924). Apologies, I just meant to request a type introduction, not logic change. Please test the changes and fight back if they break something. I'll put the logic back now, my mistake. 
I guess this shows though that the unit test introduced doesn't cover this case correctly too. It passed while issue persisted in last commit. 😅